Add `taskId` to `ActionContext` as well as `NunjuckContext`

fabiovincenzi opened the pull request:
## Hey, I just made a Pull Request!

<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->
Related Issue: https://github.com/backstage/backstage/issues/25553

This PR adds the `taskId` variable to the template context in the scaffolder-backend plugin. 
With this change, you can now access` ${{ context.task.id }}` in YAML templates, enabling task identification and tracking within workflows.

Key Changes:

- New Feature: Added taskId to the template context.
- Testing: Added unit tests to verify taskId availability in the templated context.

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

benjdlambert commented:
Think that this is looking pretty good, just wondered if you fancied maybe adding this to the `ActionContext` too under here: https://github.com/backstage/backstage/blob/7bedf44bf6541a0b7eaeffb481a0ed4912a6bc55/plugins/scaffolder-node/src/actions/types.ts#L61

So that you can access the `taskId` in `ctx.task.id` or something.

benjdlambert commented:
Nice! Thanks for the changes here, wanna bring this up at the scaffolder SIG on Friday just to review the API a little bit just to talk about the wrapping up and the ideas for what we can put under that object in the future, or if we should just make it a bit shallower in the meantime.  Feel free to join the meeting to join in on the discussion too :tada: It's at 16CET [meet.google.com/gxw-vghq-fvs](https://www.google.com/url?q=https://meet.google.com/gxw-vghq-fvs?hs%3D122%26authuser%3D0&sa=D&source=calendar&ust=1734333750675765&usg=AOvVaw0C1whAr80ZxEBFYqGcoaPU) on Friday.

benjdlambert commented on .changeset/famous-dryers-protect.md:
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
```
```suggestion
'@backstage/plugin-scaffolder-backend': minor
'@backstage/plugin-scaffolder-node': patch
```

You can delete the other changeset then :)

benjdlambert reviewed (changes requested):
Thanks for coming to the SIG, sorry that we didn't get to this until the end.

Consensus is that this is fine, just a few small things to fix up, and then we're good to ship this! :tada:

fabiovincenzi commented:
> Thanks for coming to the SIG, sorry that we didn't get to this until the end.
> 
> Consensus is that this is fine, just a few small things to fix up, and then we're good to ship this! 🎉

Sorry, I had a scheduling conflict and needed to jump off. I've addressed the suggestions — let me know how it looks now. Thanks! ☺️

benjdlambert reviewed (approved):
Thanks for this! :pray:
